src/execution/experimental-executions.js

```diff
@@ -183,126 +186,276 @@ const getMetricTitle = (suffix, title) => {
 
 const getBaseMetricTitle = partial(getMetricTitle, '');
 
-const POP_SUFFIX = ' - previous year';
-const getPoPMetricTitle = partial(getMetricTitle, POP_SUFFIX);
-
 const CONTRIBUTION_METRIC_FORMAT = '#,##0.00%';
 
+function getDefinition(measure) {
+    return get(measure, ['definition', 'measureDefinition'], {});
+}
+
+function getPoPDefinition(measure) {
+    return get(measure, ['definition', 'popMeasureDefinition'], {});
+}
+
+function getMeasureFilters(measure) {
+    return get(getDefinition(measure), 'filters', []);
+}
+
+function getAggregation(measure) {
+    return get(getDefinition(measure), 'aggregation', '').toLowerCase();
+}
+
+
+function isEmptyFilter(metricFilter) {
+    if (get(metricFilter, 'positiveAttributeFilter')) {
+        return isEmpty(get(metricFilter, ['positiveAttributeFilter', 'in']));
+    }
+    if (get(metricFilter, 'negativeAttributeFilter')) {
+        return isEmpty(get(metricFilter, ['negativeAttributeFilter', 'notIn']));
+    }
+    if (get(metricFilter, 'absoluteDateFilter')) {
+        return get(metricFilter, ['absoluteDateFilter', 'from']) === undefined && get(metricFilter, ['absoluteDateFilter', 'to']) === undefined;
+    }
+    return get(metricFilter, ['relativeDateFilter', 'from']) === undefined && get(metricFilter, ['relativeDateFilter', 'to']) === undefined;
+}
+
 const allFiltersEmpty = item => every(map(
-    get(item, 'measureFilters', []),
-    f => isEmpty(get(f, 'listAttributeFilter.default.attributeElements', []))
+    getMeasureFilters(item),
+    f => isEmptyFilter(f)
 ));
 
 const isDerived = (measure) => {
-    const type = get(measure, 'type');
-    return (type === 'fact' || type === 'attribute' || !allFiltersEmpty(measure));
+    const aggregation = getAggregation(measure);
+    return (aggregation !== '' || !allFiltersEmpty(measure));
 };
 
-const getFilterExpression = (listAttributeFilter) => {
-    const attributeUri = get(listAttributeFilter, 'listAttributeFilter.attribute');
-    const elements = get(listAttributeFilter, 'listAttributeFilter.default.attributeElements', []);
+function isAttrMeasureFilter(measureFilter) {
+    return get(measureFilter, 'positiveAttributeFilter') || get(measureFilter, 'negativeAttributeFilter');
```

Comment:
I would expect the function will return boolean value. This will return either `positiveAttributeFilter` or `negativeAttributeFilter`.

Reply:
fixed